tools/UpdateModules.ps1

```diff
@@ -26,34 +26,151 @@ function Create-ModulePsm1
+     {
+         foreach ($dll in $ModuleMetadata.NestedModules)
+         {
+             $importedModules += "Import-Module (Join-Path -Path `$PSScriptRoot -ChildPath " + $dll.Substring(2) + ")`r`n"
+         }
      }
+
      $template = Get-Content -Path $TemplatePath
-     $template = $template -replace "%MODULE-NAME%", $module.Name
+     $template = $template -replace "%MODULE-NAME%", $file.BaseName
      $template = $template -replace "%DATE%", [string](Get-Date)
-     $template = $template -replace "%STRICT-DEPENDENCIES%", $strict
-     $template = $template -replace "%DEPENDENCIES%", $loose
+     $template = $template -replace "%IMPORTED-DEPENDENCIES%", $importedModules
+
+     $contructedCommands = Find-DefaultResourceGroupCmdlets -AddDefaultParameters $AddDefaultParameters -ModuleMetadata $ModuleMetadata -ModulePath $ModulePath
+     $template = $template -replace "%COMMANDS%", $contructedCommands
+     
      Write-Host "Writing psm1 manifest to $templateOutputPath"
      $template | Out-File -FilePath $templateOutputPath -Force
      $file = Get-Item -Path $templateOutputPath
   }
 }
 
+function Find-DefaultResourceGroupCmdlets
+{
+    [CmdletBinding()]
+    param(
+        [bool]$AddDefaultParameters,
+        [Hashtable]$ModuleMetadata,
+        [string]$ModulePath
+    )
+    PROCESS
+    {
+        if ($AddDefaultParameters) 
+        {
+        $nestedModules = $ModuleMetadata.NestedModules
+        $AllCmdlets = @()
+        $nestedModules | ForEach-Object {
+            $dllPath = Join-Path -Path $ModulePath -ChildPath $_
+            $Assembly = [Reflection.Assembly]::LoadFrom($dllPath)
+            $dllCmdlets = $Assembly.GetTypes() | Where-Object {$_.CustomAttributes.AttributeType.Name -contains "CmdletAttribute"}
+            $AllCmdlets += $dllCmdlets
+        }
+        
+        $FilteredCommands = $AllCmdlets | Where-Object {Test-RequiredParameter -Cmdlet $_}
+    
+        if ($FilteredCommands.Length -eq 0) {
+            $contructedCommands = "@()"
+        }
+        else {
+            $contructedCommands = "@("
+            $FilteredCommands | ForEach-Object {
+                $contructedCommands += "'" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").VerbName + "-" + $_.GetCustomAttributes("System.Management.Automation.CmdletAttribute").NounName + ":ResourceGroupName" + "',"
+            }
+            $contructedCommands = $contructedCommands -replace ".$",")"
+        }
+    
+        return $contructedCommands
+        }
+
+        else {
+        return "@()"
+        }
+    }
+}
+
+function Test-RequiredParameter
+{
+    [CmdletBinding()]
+    param(
+        [Object]$Cmdlet
+    )
+
+    PROCESS
+    {
+        $rgParameter = $Cmdlet.GetProperties() | Where-Object {$_.Name -eq "ResourceGroupName"}
+        if ($rgParameter -ne $null) {
+            $parameterSets = $rgParameter.CustomAttributes | Where-Object {$_.AttributeType.Name -eq "ParameterAttribute"}
```

Comment:
I think this variable name is a bit confusing - I thought this was an enumeration of parameter sets, but it is really just anenumeration of parameter attributes.  Renaming this to '$parameters' or '$parameterAttributes' would make the code a bit easier to read, I think.

Reply:
done